openedx/features/journals/views/marketing.py

```diff
@@ -28,6 +36,39 @@ def bundle_about(request, bundle_uuid):
     return render_to_response('journals/bundle_about.html', context)
 
 
+def render_xblock_by_journal_access(request, usage_key_string):
+    """
+    Its a wrapper function for lms.djangoapps.courseware.views.views.render_xblock.
+    It disables 'check_if_enrolled' flag by checking that user has access on journal.
+    """
+    user_access = False
+    date_format = '%Y-%m-%d'
+    journal_uuid = request.GET.get('journal_uuid')
+    block_id = UsageKey.from_string(usage_key_string).block_id
+    cache_key = XBLOCK_JOURNAL_ACCESS_KEY.format(
+        username=request.user.username,
+        journal_uuid=journal_uuid,
+        block_id=block_id
+    )
+    journal_access_data = cache.get(cache_key)
+    if not journal_access_data:
+        journal_access_data = fetch_journal_access(
+            request.site,
+            request.user,
+            block_id=block_id
+        )
+        cache.set(cache_key, journal_access_data, XBLOCK_JOURNAL_ACCESS_KEY_TIMEOUT)
+    for journal_access in journal_access_data:
```

Comment:
I think we should processed value in cache instead of processing it after pulling from cache. We need to move that logic into a separate method.